### PR TITLE
fix(chat): fix select all files folder (Issue #775)

### PR DESCRIPTION
**Pull request by mikitabut**

### CODE REVIEW

**Comment by mikitabut** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/448258)

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-800.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-800.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/ChangePathDialog.tsx [hunk 1]
```diff
@@ -198,12 +198,13 @@ export const ChangePathDialog = ({
             newAddedFolderId: newFolderId,
             featureType: FeatureType.File,
           }}
-          handleToggleFolder={handleToggleFolder}
+          handleFolderSelect={handleFolderSelect}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChangePathDialog.tsx:202]: we still need to handle `toggle`, not only `select/open`. Especially cases `setIsAllFoldersOpened`

File: apps/chat/src/components/Common/SelectFolder/SelectFolderList.tsx [hunk 2]
```diff
@@ -19,22 +19,24 @@ import Folder, { FolderProps } from '@/src/components/Folder/Folder';
 
 interface Props<T, P = unknown> {
   folderProps: Omit<FolderProps<T, P>, 'currentFolder'>;
-  handleToggleFolder: (folderId?: string) => void;
+  handleFolderSelect: (folderId?: string) => void;
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/SelectFolder/SelectFolderList.tsx]: `folderId` can be undefined
```suggestion
  handleToggleFolder: (folderId: string) => void;
```

File: apps/chat/src/components/Chat/ChangePathDialog.tsx [hunk 3]
```diff
@@ -57,7 +57,7 @@ export const ChangePathDialog = ({
   const [isAllFoldersOpened, setIsAllFoldersOpened] = useState(true);
   const [openedFoldersIds, setOpenedFoldersIds] = useState<string[]>([]);
   const [selectedFolderId, setSelectedFolderId] = useState<string | undefined>(
-    '',
+    getRootId(),
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChangePathDialog.tsx]: is `ChangePathDialog` used only for `files`?

File: apps/chat/src/components/Files/SelectFolderModal.tsx [hunk 4]
```diff
@@ -117,10 +119,11 @@ export const SelectFolderModal = ({
             loadingFolderIds: loadingFolderIds,
             featureType: FeatureType.File,
           }}
-          handleToggleFolder={handleToggleFolder}
+          handleFolderSelect={handleFolderSelect}
           isAllEntitiesOpened={isAllFilesOpened}
           selectedFolderId={selectedFolderId}
           rootFolderName="All files"
+          rootFolderId={getRootId()}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Files/SelectFolderModal.tsx]: is `SelectFolderModal` used only for `files`?

File: apps/chat/src/components/Files/SelectFolderModal.tsx [hunk 5]
```diff
@@ -84,7 +86,7 @@ export const SelectFolderModal = ({
   );
 
   const handleFolderSelect = useCallback(
-    (folderId: string) => {
+    (folderId?: string) => {
```
Comment (IlyaBondar) [inline apps/chat/src/components/Files/SelectFolderModal.tsx]: why can `folderId` be `undefined`?

File: apps/chat/src/components/Files/SelectFolderModal.tsx (hunk 4 above)
Comment (mikitabut) [reply apps/chat/src/components/Files/SelectFolderModal.tsx:126]: Yes

File: apps/chat/src/components/Chat/ChangePathDialog.tsx (hunk 3 above)
Comment (mikitabut) [reply apps/chat/src/components/Chat/ChangePathDialog.tsx]: Yes

File: apps/chat/src/components/Files/SelectFolderModal.tsx (hunk 4 above)
Comment (mikitabut) [reply apps/chat/src/components/Files/SelectFolderModal.tsx]: Yes

File: apps/chat/src/components/Chat/ChangePathDialog.tsx (hunk 1 above)
Comment (mikitabut) [reply apps/chat/src/components/Chat/ChangePathDialog.tsx:202]: Actually toggle handled inside select folder
Moreover you cannot toggle without selecting

File: apps/chat/src/components/Files/SelectFolderModal.tsx (hunk 5 above)
Comment (mikitabut) [reply apps/chat/src/components/Files/SelectFolderModal.tsx]: Removed

File: apps/chat/src/components/Common/SelectFolder/SelectFolderList.tsx (hunk 2 above)
Comment (mikitabut) [reply apps/chat/src/components/Common/SelectFolder/SelectFolderList.tsx]: It cannot be undefined so removed '?'

**Review by Alexander-Kezik — APPROVED** [review]

**Pull request merged**